### PR TITLE
test: Moved kfp-kubernetes-execution-tests to GitHub Actions

### PR DESCRIPTION
**Pull request by hbelmiro**
**Description of your changes:**
Moved `kfp-kubernetes-execution-tests` to GitHub Actions.

Resolves: https://github.com/kubeflow/pipelines/issues/10921

### Implementation notes

- I had to use Ubuntu 24.04 because the default (22.04) has an old and non-compatible version of `protobuf-compiler`.
  - `apt-get update` was not enough to get it updated
  - I also tried to manually install the binary, but it didn't work
  - Python 3.8 is not available in the Ubuntu 24.04 runner. So I used 3.9.

### PR to update GoogleCloudPlatform/oss-test-infra

- https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2304

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by rimolive** [comment]
@hbelmiro Prow tests run `kfp-kubernetes` testing using Python 3.8 to 3.12. From what I can see in this PR, GHA `kfp-kubernetes` testing is not checking Python 3.8. Can you add this version too?

**Comment by hbelmiro** [comment]
> @hbelmiro Prow tests run `kfp-kubernetes` testing using Python 3.8 to 3.12. From what I can see in this PR, GHA `kfp-kubernetes` testing is not checking Python 3.8. Can you add this version too?
> 
> 

@rimolive I added a note about that in the description of the PR.

**Review by DharmitD — COMMENTED** [review]
/lgtm

@hbelmiro could you make a PR to the oss-test-infra repo in parallel to remove these tests from Prow?

@chensun could you take a look? Thanks.

**Comment by hbelmiro** [comment]
cc @chensun @zijianjoy @james-jwu

**Review by DharmitD — APPROVED** [review]
/Approve

File: .github/workflows/kfp-kubernetes-execution-tests.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        with:
+          cluster_name: kfp
+          kubectl_version: v1.29.2
+          version: v0.22.0
+          node_image: kindest/node:v1.29.2
+
+      - name: Build images
+        run: ./scripts/deploy/github/build-images.sh
+
+      - name: Deploy KFP
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply .github/workflows/kfp-kubernetes-execution-tests.yml]: Good catch! I thought I had updated that.
Updated now. Thank you.

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Comment by hbelmiro** [comment]
> Needs approval from an approver in each of these files:
> 
>     * ~[.github/OWNERS](https://github.com/kubeflow/pipelines/blob/master/.github/OWNERS)~ [DharmitD,hbelmiro]
> 
>     * **[kubernetes_platform/OWNERS](https://github.com/kubeflow/pipelines/blob/master/kubernetes_platform/OWNERS)**
> 
>     * ~[test/OWNERS](https://github.com/kubeflow/pipelines/blob/master/test/OWNERS)~ [DharmitD,hbelmiro]

@connor-mccarthy we need approval from one of you guys.

File: .github/workflows/kfp-kubernetes-execution-tests.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        with:
+          python-version: '3.9'
+
+      - name: apt-get update
+        run: sudo apt-get update
+
+      - name: Install protobuf-compiler
+        run: sudo apt-get install protobuf-compiler -y
+
+      - name: Install setuptools
```
Comment (diegolovison) [inline .github/workflows/kfp-kubernetes-execution-tests.yml:38]: We need to create scripts and call them from GitHub Actions (GHA). This way, we can run the test suite independently of GitHub. Many vendors depend on their infrastructure to certify releases. Additionally, this approach enables component reuse within GHA.

**Review by diegolovison — CHANGES_REQUESTED** [review]

File: .github/workflows/kfp-kubernetes-execution-tests.yml (hunk 2 above)
Comment (hbelmiro) [reply .github/workflows/kfp-kubernetes-execution-tests.yml:38]: Agreed. We can take that approach in all tests.
Can we do that in a follow-up effort after the migration to GHA? It's quite important to have the tests running in GHA ASAP.

cc @DharmitD @HumairAK

Comment (diegolovison) [reply .github/workflows/kfp-kubernetes-execution-tests.yml:38]: Ok!

**Comment by hbelmiro** [comment]
@james-jwu can you help me to get this approved?

**Comment by DharmitD** [comment]
@chensun could you please approve? Thank you!

**Review by DharmitD — APPROVED** [review]
/Approve

**Comment by hbelmiro** [comment]
@DharmitD can you re-lgtm?

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Pull request merged**